primitives/message-lane/src/lib.rs

```diff
@@ -74,34 +74,48 @@ pub struct Message<Fee> {
 /// Inbound lane data.
 #[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq)]
 pub struct InboundLaneData<RelayerId> {
-	/// Identifiers of relayers and messages that they have delivered (ordered by message nonce).
-	/// It is guaranteed to have at most N entries, where N is configured at module level. If
-	/// there are N entries in this vec, then:
+	/// Identifiers of relayers and messages that they have delivered to this lane (ordered by message nonce).
+	///
+	/// This serves as a helper storage item, to allow the source chain to easily pay rewards
+	/// to the relayers who succesfuly delivered messages to the target chain (inbound lane).
+	///
+	/// It is guaranteed to have at most N entries, where N is configured at the module level.
+	/// If there are N entries in this vec, then:
 	/// 1) all incoming messages are rejected if they're missing corresponding `proof-of(outbound-lane.state)`;
-	/// 2) all incoming messages are rejected if `proof-of(outbound-lane.state).latest_received_nonce` is
-	///    equal to `this.latest_confirmed_nonce`.
-	/// Given what is said above, all nonces in this queue are in range (latest_confirmed_nonce; latest_received_nonce].
+	/// 2) all incoming messages are rejected if `proof-of(outbound-lane.state).last_delivered_nonce` is
+	///    equal to `self.last_confirmed_nonce`.
+	/// Given what is said above, all nonces in this queue are in range:
+	/// `(self.last_confirmed_nonce; self.last_delivered_nonce()]`.
 	///
 	/// When a relayer sends a single message, both of MessageNonces are the same.
 	/// When relayer sends messages in a batch, the first arg is the lowest nonce, second arg the highest nonce.
-	/// Multiple dispatches from the same relayer one are allowed.
+	/// Multiple dispatches from the same relayer are allowed.
 	pub relayers: VecDeque<(MessageNonce, MessageNonce, RelayerId)>,
-	/// Nonce of latest message that we have received from bridged chain.
-	pub latest_received_nonce: MessageNonce,
-	/// Nonce of latest message that has been confirmed to the bridged chain.
-	pub latest_confirmed_nonce: MessageNonce,
+
+	/// Nonce of the last message which delivery to this (target) chain has been confirmed back
+	/// on the source chain.
```

Comment:
Perhaps: "Nonce of the last source chain confirmed message, which has been delivered to this (target) chain."

Reply:
I'm not convinced it's better. In my sentence I've tried to emphasize the ordering of events as well:
1. Delivery to the target chain.
2. Delivery confirmation on the source chain.

With `source-chain-confirmed message` I feel this is lost, and it's not clear to me how that differs from `last_delivered_nonce()`.

Reply:
So how about, "Nonce of the last message that a) has been delivered to the target chain and b) for which there is a confimation on the source chain." I just felt that the original sentence reads a bit 'bumpy'.

Reply:
Re-phrased a bit and updated, let me know if the bumps are smaller now :)

Reply:
Smooth